### PR TITLE
feat: WDS Checkbox Group Integration

### LINKED ISSUE #28102: Integrate WDS CheckboxGroup component with appsmith
- [CheckboxGroup in Storybrook](https://release--64c89da059b7d62ad3a60544.chromatic.com/?path=/docs/design-system-widgets-checkboxgroup--docs)
- [CheckboxGroup code](https://github.com/appsmithorg/appsmith/tree/master/app/client/packages/design-system/widgets/src/components/CheckboxGroup)

In order to get acquainted with what properties should be in property pane, please read [this doc](https://www.notion.so/appsmith/Widgets-Properties-Audit-3832f6a7be5f47cc9bfa3bd1ddae31fa).

### PR DESCRIPTION
**Pull request by dhruvikn**
## Description
Adds WDSCheckboxGroup Widget.

#### PR fixes following issue(s)
Fixes #28102

#### Media
<video src="https://github.com/appsmithorg/appsmith/assets/22471214/78ed3ff5-0b60-4ba1-8a91-1b31f7e30334"></video>

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [x] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Review by KelvinOm — APPROVED** [review]

File: app/client/src/widgets/wds/WDSCheckboxGroupWidget/component/index.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import type { CheckboxGroupComponentProps } from "./types";
+import type { OptionProps } from "../widget/types";
+
+export const CheckboxGroupComponent = (props: CheckboxGroupComponentProps) => {
+  return (
+    <CheckboxGroup
+      isDisabled={props.isDisabled}
+      isRequired={props.isRequired}
+      label={props.labelText}
+      onChange={(value) => {
```
Comment (jsartisan) [inline app/client/src/widgets/wds/WDSCheckboxGroupWidget/component/index.tsx:12]: you can simply do `onChange={props.onChange}`

that should work too.

**Review by jsartisan — APPROVED** [review]
LGTM, left a minor comment.

File: app/client/src/widgets/wds/WDSCheckboxGroupWidget/component/index.tsx (hunk 1 above)
Comment (dhruvikn) [reply app/client/src/widgets/wds/WDSCheckboxGroupWidget/component/index.tsx:12]: Oh yes, thanks for this. I was debugging and added this for a breakpoint and forgot to change this.

**Comment by dhruvikn** [comment]
@KelvinOm @jsartisan @ichik - I'm not adding the "select all" property right now. I have looked into how we can do this effectively and I couldn't come up with an intuitive solution (in terms of writing a non-hacky code). 

I also think we can introduce this property at the WDS component level rather than the widget level. I have raised another ticket for this here - https://github.com/appsmithorg/appsmith/issues/28581

Let me know what you folks think.

**Review by jsartisan — APPROVED** [review]

**Review by KelvinOm — APPROVED** [review]

**Pull request merged**